--- a/arch/arm/boot/dts/microchip/at91-sama5d29_curiosity.dts
+++ b/arch/arm/boot/dts/microchip/at91-sama5d29_curiosity.dts
@@ -133,9 +133,21 @@
 
 	spi6: spi@400 {
 		dmas = <0>, <0>;
+		compatible = "atmel,at91rm9200-spi";
+		reg = <0x400>;
+		#address-cells = <1>;
+		#size-cells = <0>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_rpi_spi>;
 		status = "okay";
+
+		tpm@0 {
+			compatible = "infineon,slb9670", "tcg,tpm_tis-spi";
+			reg = <0>;
+			spi-max-frequency = <10000000>;
+			status = "okay";
+		};
+
 	};
 };
 
@@ -401,8 +413,7 @@
 		pinmux = <PIN_PD12__FLEXCOM4_IO0>,
 			 <PIN_PD13__FLEXCOM4_IO1>,
 			 <PIN_PD14__FLEXCOM4_IO2>,
-			 <PIN_PD15__FLEXCOM4_IO3>,
-			 <PIN_PD16__FLEXCOM4_IO4>;
+			 <PIN_PD15__FLEXCOM4_IO3>;
 		bias-disable;
 	};
 
