--- a/drivers/char/tpm/tpm_tis_spi_main.c
+++ b/drivers/char/tpm/tpm_tis_spi_main.c
@@ -86,7 +86,7 @@
 	struct spi_message m;
 	u8 transfer_len;
 	int ret;
-
+	printk(KERN_INFO "Entered tpm_tis_spi_transfer_hald\n");
 	while (len) {
 		transfer_len = min_t(u16, len, MAX_SPI_FRAMESIZE);
 
@@ -124,6 +124,7 @@
 		reinit_completion(&phy->ready);
 
 		ret = spi_sync(phy->spi_device, &m);
+		printk(KERN_INFO "spi_sync returned %d\n",ret);
 		if (ret < 0)
 			return ret;
 
@@ -134,7 +135,7 @@
 
 		len -= transfer_len;
 	}
-
+	printk(KERN_INFO "tpm_tis_spi_transfer_half is returning %d\n",ret);
 	return ret;
 }
 
@@ -148,7 +149,7 @@
 	u8 transfer_len;
 
 	spi_bus_lock(phy->spi_device->controller);
-
+	printk(KERN_INFO "Entered tpm_tis_spi_transfer_fulll\n");
 	while (len) {
 		transfer_len = min_t(u16, len, MAX_SPI_FRAMESIZE);
 
@@ -162,16 +163,20 @@
 		spi_xfer.rx_buf = phy->iobuf;
 		spi_xfer.len = 4;
 		spi_xfer.cs_change = 1;
-
+		printk(KERN_INFO "spi_message_init is being called\n");
 		spi_message_init(&m);
+		printk(KERN_INFO "spi_message_add_tail is being called\n");
 		spi_message_add_tail(&spi_xfer, &m);
+		printk(KERN_INFO "spi_sync_locked being called\n");
 		ret = spi_sync_locked(phy->spi_device, &m);
+		printk(KERN_INFO "spi_sync_locked returned %d\n",ret);
 		if (ret < 0)
 			goto exit;
 
 		/* Flow control transfers are receive only */
 		spi_xfer.tx_buf = NULL;
 		ret = phy->flow_control(phy, &spi_xfer);
+		printk(KERN_INFO "spi_flow control returned %d\n",ret);
 		if (ret < 0)
 			goto exit;
 
@@ -191,6 +196,7 @@
 		spi_message_add_tail(&spi_xfer, &m);
 		reinit_completion(&phy->ready);
 		ret = spi_sync_locked(phy->spi_device, &m);
+		printk(KERN_INFO "spi_sync_locked returned %d\n",ret);
 		if (ret < 0)
 			goto exit;
 
@@ -210,6 +216,7 @@
 		spi_message_add_tail(&spi_xfer, &m);
 		spi_sync_locked(phy->spi_device, &m);
 	}
+	printk(KERN_INFO "full returning %d\n",ret);
 
 	spi_bus_unlock(phy->spi_device->controller);
 	return ret;
@@ -227,6 +234,7 @@
 	 * wait polling in controller.
 	 * Set TPM HW flow control flag..
 	 */
+	printk(KERN_INFO "Entered tpm_tis_spi_transfer\n");
 	if (ctlr->flags & SPI_CONTROLLER_HALF_DUPLEX)
 		return tpm_tis_spi_transfer_half(data, addr, len, in, out);
 	else
@@ -236,6 +244,7 @@
 static int tpm_tis_spi_read_bytes(struct tpm_tis_data *data, u32 addr,
 				  u16 len, u8 *result, enum tpm_tis_io_mode io_mode)
 {
+	printk(KERN_INFO "Entered tpm_tis_spi_read_bytes\n");
 	return tpm_tis_spi_transfer(data, addr, len, result, NULL);
 }
 
@@ -251,7 +260,7 @@
 	phy->iobuf = devm_kmalloc(&spi->dev, SPI_HDRSIZE + MAX_SPI_FRAMESIZE, GFP_KERNEL);
 	if (!phy->iobuf)
 		return -ENOMEM;
-
+	printk(KERN_INFO "About to call tpm_tis_core_init\n");
 	phy->spi_device = spi;
 
 	return tpm_tis_core_init(&spi->dev, &phy->priv, irq, phy_ops, NULL);
@@ -266,7 +275,7 @@
 {
 	struct tpm_tis_spi_phy *phy;
 	int irq;
-
+	printk(KERN_INFO "Entering tpm_tis_spi_main probe function%s\n",dev_name(&dev->dev));
 	phy = devm_kzalloc(&dev->dev, sizeof(struct tpm_tis_spi_phy),
 			   GFP_KERNEL);
 	if (!phy)
